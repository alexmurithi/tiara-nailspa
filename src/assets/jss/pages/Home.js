import { makeStyles } from 'tss-react/mui'

const homeStyles = makeStyles()(theme => {
  return {
    heroWrapper: {
      width: '100%',
      height: '680px',
      position: 'relative',
      color: theme.palette.common.white,
    },
    heroGradient: {
      position: 'absolute',
      inset: 0,
      backgroundImage: 'linear-gradient(to bottom, #DB2777, #FFFFFF)',
      opacity: 0.4,
    },
    heroContentWrapper: {
      [theme.breakpoints.down('md')]: {
        marginTop: theme.spacing(8),
      },
      marginTop: theme.spacing(14),
      textAlign: 'center',
    },
    heroContentDescription: {
      [theme.breakpoints.down('md')]: {
        fontSize: '14px',
        fontWeight: '300',
      },
    },
    heroContentH1: {
      textTransform: 'capitalize',
      [theme.breakpoints.down('md')]: {
        fontSize: '20px',
      },
    },
    heroCard: {
      background: 'transparent',
      border: `2px solid ${theme.palette.common.white}`,
      borderRadius: '12px',
      color: theme.palette.common.white,
      width: '320px',
      height: '200px',
      [theme.breakpoints.down('md')]: {
        overflowY: 'scroll',
        width: '100%',
        height: '100px',
      },
    },
    heroCardContentStack: {
      justifyContent: 'space-between',
      alignItems: 'center',
    },
    heroCardContentText: {
      [theme.breakpoints.down('md')]: {
        fontSize: '14px',
      },
    },
    heroBookAppointmentBtnWrapper: {
      margin: `${theme.spacing(6)} 0 !important`,
    },
    heroBookAppointmentBtn: {
      color: theme.palette.common.white,
      borderRadius: 0,
      width: '375px',
      padding: theme.spacing(2),
      [theme.breakpoints.down('md')]: {
        width: '100%',
      },
    },
    heroBookAppointmentBtnText: {
      fontWeight: 'bold',
    },
  }
})

export default homeStyles
