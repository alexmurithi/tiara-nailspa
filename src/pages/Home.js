import React from 'react'
import {
  Box,
  Container,
  Typography,
  Stack,
  Grid,
  Card,
  CardContent,
  Button,
} from '@mui/material'
import LocationOnOutlinedIcon from '@mui/icons-material/LocationOnOutlined'
import AccessTimeOutlinedIcon from '@mui/icons-material/AccessTimeOutlined'
import ContactsOutlinedIcon from '@mui/icons-material/ContactsOutlined'
import BackgroundImage from '@images/desktop/background.jpg'
import Image from '@components/Image'
import homeStyles from '@styles/pages/Home'

const Home = () => {
  const { classes } = homeStyles()
  return (
    <>
      <Box className={classes.heroWrapper}>
        <Box className={classes.heroGradient} />
        <Image
          src={BackgroundImage}
          width="100%"
          height="100%"
          alt="Tiara Background"
        />
        <Box
          sx={{
            position: 'absolute',
            inset: 0,
          }}
        >
          <Container maxWidth="lg">
            <Stack
              direction="column"
              spacing={{ xs: 2 }}
              className={classes.heroContentWrapper}
            >
              <Typography
                variant="h1"
                gutterBottom
                className={classes.heroContentH1}
              >
                Welcome to tiara nail spa
              </Typography>
              <Typography
                variant="body1"
                fontWeight="200"
                gutterBottom
                className={classes.heroContentDescription}
              >
                Let us give your nails a treatment!
              </Typography>
              <Grid container rowSpacing={2} columnSpacing={{ md: 4 }}>
                <Grid item xs={12} md={4}>
                  <Card elevation={0} className={classes.heroCard}>
                    <CardContent>
                      <Stack
                        direction={{ xs: 'row', lg: 'column' }}
                        spacing={2}
                        className={classes.heroCardContentStack}
                      >
                        <Box>
                          <LocationOnOutlinedIcon fontSize="large" />
                        </Box>
                        <Box>
                          <Typography
                            fontWeight="bold"
                            className={classes.heroCardContentText}
                            variant="body1"
                          >
                            Prime Plaza, 3rd Floor, Room 401, Nakuru
                          </Typography>
                        </Box>
                      </Stack>
                    </CardContent>
                  </Card>
                </Grid>
                <Grid item xs={12} md={4}>
                  <Card elevation={0} className={classes.heroCard}>
                    <CardContent>
                      <Stack
                        direction={{ xs: 'row', lg: 'column' }}
                        spacing={2}
                        className={classes.heroCardContentStack}
                      >
                        <Box>
                          <AccessTimeOutlinedIcon fontSize="large" />
                        </Box>
                        <Box>
                          <Typography
                            fontWeight="bold"
                            className={classes.heroCardContentText}
                            variant="body1"
                            gutterBottom
                          >
                            Monday - Friday : 09:00 - 17:00
                          </Typography>
                          <Typography
                            fontWeight="bold"
                            className={classes.heroCardContentText}
                            variant="body1"
                            gutterBottom
                          >
                            Saturday : 11:00 - 17:00
                          </Typography>
                          <Typography
                            fontWeight="bold"
                            className={classes.heroCardContentText}
                            variant="body1"
                            gutterBottom
                          >
                            Sunday : 14:00 - 17:00
                          </Typography>
                        </Box>
                      </Stack>
                    </CardContent>
                  </Card>
                </Grid>
                <Grid item xs={12} md={4}>
                  <Card elevation={0} className={classes.heroCard}>
                    <CardContent>
                      <Stack
                        direction={{ xs: 'row', lg: 'column' }}
                        spacing={2}
                        className={classes.heroCardContentStack}
                      >
                        <Box>
                          <ContactsOutlinedIcon fontSize="large" />
                        </Box>
                        <Box>
                          <Typography
                            fontWeight="bold"
                            className={classes.heroCardContentText}
                            variant="body1"
                            gutterBottom
                          >
                            tiara@nailspa.com
                          </Typography>
                          <Typography
                            fontWeight="bold"
                            className={classes.heroCardContentText}
                            variant="body1"
                            gutterBottom
                          >
                            +254712293003
                          </Typography>
                        </Box>
                      </Stack>
                    </CardContent>
                  </Card>
                </Grid>
              </Grid>
              <Box className={classes.heroBookAppointmentBtnWrapper}>
                <Button
                  variant="contained"
                  color="primary"
                  className={classes.heroBookAppointmentBtn}
                  disableElevation
                  disableRipple
                >
                  <Typography
                    variant="subtitle2"
                    className={classes.heroBookAppointmentBtnText}
                  >
                    Book Appointment Now!
                  </Typography>
                </Button>
              </Box>
            </Stack>
          </Container>
        </Box>
      </Box>
    </>
  )
}

export default React.memo(Home)
