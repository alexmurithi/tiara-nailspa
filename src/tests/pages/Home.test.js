/* eslint-disable no-undef */
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'
import { ThemeProvider } from '@mui/material'
import theme from '@themes/Default'
import Home from '@pages/Home'

describe('<Home/>', () => {
  it('renders corretly without crashing', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Home />
        </ThemeProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
