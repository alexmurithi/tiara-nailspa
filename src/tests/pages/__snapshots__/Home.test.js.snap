// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> renders corretly without crashing 1`] = `
<div
  className="MuiBox-root css-1umzs9j-heroWrapper"
>
  <div
    className="MuiBox-root css-17o8e0z-heroGradient"
  />
  <img
    alt="Taiara Nail Spa"
    className="css-13glzxp-image"
    height="100%"
    src="background.jpg"
    width="100%"
  />
  <div
    className="MuiBox-root css-1ayv084"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        className="MuiStack-root css-rj6z5e-MuiStack-root-heroContentWrapper"
      >
        <h1
          body1="span"
          body2="span"
          className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-1npm43r-MuiTypography-root-heroContentH1"
          h1="h1"
          h2="h2"
          h3="h3"
          h4="h4"
          h5="h5"
          h6="h6"
          subtitle1="p"
          subtitle2="p"
        >
          Welcome to tiara nail spa
        </h1>
        <p
          body1="span"
          body2="span"
          className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1m265ue-MuiTypography-root-heroContentDescription"
          h1="h1"
          h2="h2"
          h3="h3"
          h4="h4"
          h5="h5"
          h6="h6"
          subtitle1="p"
          subtitle2="p"
        >
          Let us give your nails a treatment!
        </p>
        <div
          className="MuiGrid-root MuiGrid-container css-1q06fmj-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-185f7v7-MuiPaper-root-MuiCard-root-heroCard"
            >
              <div
                className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  className="MuiStack-root css-hucc60-MuiStack-root-heroCardContentStack"
                >
                  <div
                    className="MuiBox-root css-0"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                      data-testid="LocationOnOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
                      />
                      <circle
                        cx="12"
                        cy="9"
                        r="2.5"
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiBox-root css-0"
                  >
                    <p
                      body1="span"
                      body2="span"
                      className="MuiTypography-root MuiTypography-body1 css-1bhnmwp-MuiTypography-root-heroCardContentText"
                      h1="h1"
                      h2="h2"
                      h3="h3"
                      h4="h4"
                      h5="h5"
                      h6="h6"
                      subtitle1="p"
                      subtitle2="p"
                    >
                      Prime Plaza, 3rd Floor, Room 401, Nakuru
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-185f7v7-MuiPaper-root-MuiCard-root-heroCard"
            >
              <div
                className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  className="MuiStack-root css-hucc60-MuiStack-root-heroCardContentStack"
                >
                  <div
                    className="MuiBox-root css-0"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                      data-testid="AccessTimeOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiBox-root css-0"
                  >
                    <p
                      body1="span"
                      body2="span"
                      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1gc1f5b-MuiTypography-root-heroCardContentText"
                      h1="h1"
                      h2="h2"
                      h3="h3"
                      h4="h4"
                      h5="h5"
                      h6="h6"
                      subtitle1="p"
                      subtitle2="p"
                    >
                      Monday - Friday : 09:00 - 17:00
                    </p>
                    <p
                      body1="span"
                      body2="span"
                      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1gc1f5b-MuiTypography-root-heroCardContentText"
                      h1="h1"
                      h2="h2"
                      h3="h3"
                      h4="h4"
                      h5="h5"
                      h6="h6"
                      subtitle1="p"
                      subtitle2="p"
                    >
                      Saturday : 11:00 - 17:00
                    </p>
                    <p
                      body1="span"
                      body2="span"
                      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1gc1f5b-MuiTypography-root-heroCardContentText"
                      h1="h1"
                      h2="h2"
                      h3="h3"
                      h4="h4"
                      h5="h5"
                      h6="h6"
                      subtitle1="p"
                      subtitle2="p"
                    >
                      Sunday : 14:00 - 17:00
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-185f7v7-MuiPaper-root-MuiCard-root-heroCard"
            >
              <div
                className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  className="MuiStack-root css-hucc60-MuiStack-root-heroCardContentStack"
                >
                  <div
                    className="MuiBox-root css-0"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                      data-testid="ContactsOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V6h16v12zM4 0h16v2H4zm0 22h16v2H4zm8-10c1.38 0 2.5-1.12 2.5-2.5S13.38 7 12 7 9.5 8.12 9.5 9.5 10.62 12 12 12zm0-3.5c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm5 7.49C17 13.9 13.69 13 12 13s-5 .9-5 2.99V17h10v-1.01zm-8.19-.49c.61-.52 2.03-1 3.19-1 1.17 0 2.59.48 3.2 1H8.81z"
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiBox-root css-0"
                  >
                    <p
                      body1="span"
                      body2="span"
                      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1gc1f5b-MuiTypography-root-heroCardContentText"
                      h1="h1"
                      h2="h2"
                      h3="h3"
                      h4="h4"
                      h5="h5"
                      h6="h6"
                      subtitle1="p"
                      subtitle2="p"
                    >
                      tiara@nailspa.com
                    </p>
                    <p
                      body1="span"
                      body2="span"
                      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1gc1f5b-MuiTypography-root-heroCardContentText"
                      h1="h1"
                      h2="h2"
                      h3="h3"
                      h4="h4"
                      h5="h5"
                      h6="h6"
                      subtitle1="p"
                      subtitle2="p"
                    >
                      +254712293003
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiBox-root css-1qtw31u-heroBookAppointmentBtnWrapper"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-1ijvlkj-MuiButtonBase-root-MuiButton-root-heroBookAppointmentBtn"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <h6
              body1="span"
              body2="span"
              className="MuiTypography-root MuiTypography-subtitle2 css-7r8p4b-MuiTypography-root-heroBookAppointmentBtnText"
              h1="h1"
              h2="h2"
              h3="h3"
              h4="h4"
              h5="h5"
              h6="h6"
              subtitle1="p"
              subtitle2="p"
            >
              Book Appointment Now!
            </h6>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
